src/treemodel/range.js

```diff
@@ -261,32 +252,51 @@ export default class Range {
-	 * we enter into when iterating over this range.
+	 * In other words, this iterates over all text characters that are inside the range and all the
+	 * {@link core.treeModel.Element}s we enter into when iterating over this range.
+	 *
+	 * **Note:** this method will not return a parent node of start position. This is in contrary to
+	 * {@link core.treeModel.TreeWalker} which will return that node with `'ELEMENT_END'` type. This method also
+	 * returns each {@link core.treeModel.Element} once, while simply used {@link core.treeModel.TreeWalker} might
+	 * return it twice: for `'ELEMENT_START'` and `'ELEMENT_END'`.
 	 *
-	 * **Note:** this method will not return a parent node of start position. This is in contrary to {@link core.treeModel.TreeWalker}
-	 * which will return that node with `'ELEMENT_END'` type. This method also returns each {@link core.treeModel.Element} once,
-	 * while simply used {@link core.treeModel.TreeWalker} might return it twice: for `'ELEMENT_START'` and `'ELEMENT_END'`.
+	 * **Node:*** because this method does not return {@link core.treeModel.TreeWalkerValue values} with the type of
+	 * `'ELEMENT_END'`, you can use {@link core.treeModel.TreeWalkerValue.previousPosition} as a position before the
+	 * item.
 	 *
 	 * @see {treeModel.TreeWalker}
 	 * @param {Boolean} [mergeCharacters] Flag indicating whether all consecutive characters with the same attributes
-	 * should be returned as one {@link core.treeModel.TextFragment} (`true`) or one by one as multiple {@link core.treeModel.CharacterProxy}
-	 * (`false`) objects. Defaults to `false`.
-	 * @returns {Iterable.<core.treeModel.Node|treeModel.TextFragment>}
+	 * should be returned as one {@link core.treeModel.TextFragment} (`true`) or one by one as multiple
+	 * {@link core.treeModel.CharacterProxy} (`false`) objects. Defaults to `false`.
+	 * @returns {Iterable.<core.treeModel.TreeWalkerValue>}
 	 */
-	*getAllNodes( mergeCharacters ) {
-		let it = new TreeWalker( { boundaries: this, mergeCharacters: mergeCharacters } );
-		let step;
+	*getValues( mergeCharacters ) {
+		const treeWalker = new TreeWalker( { boundaries: this, mergeCharacters: mergeCharacters } );
 
-		do {
-			step = it.next();
-
-			if ( step.value && step.value.type != 'ELEMENT_END' ) {
-				yield step.value.item;
+		for ( let value of treeWalker ) {
+			if ( value.type != 'ELEMENT_END' ) {
+				yield value;
 			}
-		} while ( !step.done );
+		}
+	}
+
+	/**
+	 * Use {@link core.treeModel.Range#getValues}, but return only {@link core.treeModel.Item items} from
+	 * {@link core.treeModel.TreeWalkerValue}.
+	 *
+	 * @see {core.treeModel.Range#getValues}
+	 * @param {Boolean} [mergeCharacters] See {@link core.treeModel.Range#getValues}.
+	 * @returns {Iterable.<core.treeModel.Item>}
+	 */
+	*getItems( mergeCharacters ) {
+		for ( let value of this.getValues( mergeCharacters ) ) {
```

Comment:
In the F2F discussion we decided that `mergeCharacters` will be usually true so should be changed to `singleCharacters`, which is false by default.